### PR TITLE
MAINT: additional test truthiness and length the empty Rotation

### LINKED ISSUE #13452: Calling `len()` on the `scipy.spatial.transform.rotation.Rotation` object with one rotation should return 1
When calling the `len()` function on the `scipy.spatial.transform.rotation.Rotation` object with exactly one rotation, it currently raises a TypeError instead of returning 1.

([Link to source code](https://github.com/scipy/scipy/blob/0cb5a06c9465b99737f914dce48d9f5aa4400e31/scipy/spatial/transform/rotation.pyx#L537))

#### Reproducing code example:

```
from scipy.spatial.transform import Rotation as R

r = R.from_quat([1, 0, 0, 0])
print(len(r))    # raises TypeError
```

#### Error message:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-227-c5f861eb1f8a> in <module>
      4 try:
----> 5     print(len(r))
      6 except Exception as e:

rotation.pyx in scipy.spatial.transform.rotation.Rotation.__len__()

TypeError: Single rotation has no len().
```

Calling `len()` on an object with 2 rotations returns 2 as expected:

```
r = R.from_quat([[1, 0, 0, 0],
                 [1, 0, 0, 0]])
print(len(r))    # returns 2
```

Moreover, when creating a rotation object using the `from_euler()` class method, it correctly returns 1 for one rotation around a single coordinate axis, but raises the above error again for a rotation around more than one coordinate axes:

```
r = R.from_euler('x', [0.1])
print(len(r))    # returns 1

r = R.from_euler('xz', [0.1, 0.1])
print(len(r))    # raises TypeError
```

#### Scipy/Numpy/Python version information:

```
import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.6.0 1.19.5 sys.version_info(major=3, minor=9, micro=1, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by fabianegli**
#### Reference issue

Closes #13452

### CODE REVIEW

**Comment by fabianegli** [comment]
Once again this would be helpful: https://github.com/community/community/discussions/15452

**Comment by rkern** [comment]
Consider also testing the `len(Rotation([[0, 0, 0, 1]])) == 1` case to clarify the difference between a "single" `Rotation` and a length-1 `Rotation`.

File: scipy/spatial/transform/tests/test_rotation.py [hunk 1]
```diff
@@ -1354,10 +1354,13 @@ def test_concatenate_wrong_type():
...
+    rotation_multi_empty = Rotation(np.empty((0, 4)))
 
     with pytest.raises(TypeError, match="Single rotation has no len()."):
         len(rotation_single)
     assert len(rotation_multi) == 2
+    assert len(rotation_multi_empty) == 0
 
     assert rotation_single
     assert rotation_multi
+    assert rotation_multi_empty  # Rotation should alwas be true. See gh-16663
```
Comment (fabianegli) [inline scipy/spatial/transform/tests/test_rotation.py:1366]:
```suggestion
    rotation_multi_empty = Rotation(np.empty((0, 4)))
    rotation_multi_one = Rotation([[0, 0, 0, 1]])
    rotation_multi = Rotation([[0, 0, 0, 1], [0, 0, 0, 1]])
    rotation_single = Rotation([0, 0, 0, 1])

    assert len(rotation_multi_empty) == 0
    assert len(rotation_multi_one) == 1
    assert len(rotation_multi) == 2
    with pytest.raises(TypeError, match="Single rotation has no len()."):
        len(rotation_single)

    # Rotation should always be truthy. See gh-16663
    assert rotation_multi_empty
    assert rotation_multi_one
    assert rotation_multi
    assert rotation_single
```

**Comment by fabianegli** [comment]
Again edditing on GH - hope it passes linting 🤞

**Review by tylerjereddy — APPROVED** [review]
I think you've addressed Robert's review comment, and this doesn't change any old tests, just adds new cases, which is pretty safe. The CI failure is unrelated. This test may be getting to the point where parametrization may help, but doesn't seem worth more adjustments at this stage.

I won't comment on whether some of these might be confusing, but it certainly makes sense to me to encode the behavior in tests, if for no other reason than to make sure that any future changes are intentional rather than accidental "drift" in the code.

**Pull request merged**